pkg/route/api/validation/validation.go

```diff
@@ -26,7 +26,8 @@ func ValidateRoute(route *routeapi.Route) field.ErrorList {
 
 	//host is not required but if it is set ensure it meets DNS requirements
 	if len(route.Spec.Host) > 0 {
-		if len(kvalidation.IsDNS1123Subdomain(route.Spec.Host)) != 0 {
+		hostname, _ := routeapi.NormalizeWildcardHost(route.Spec.Host)
+		if len(kvalidation.IsDNS1123Subdomain(hostname)) != 0 {
```

Comment:
What happens if the API admits a wildcard route but the router doesn't have wild cards enabled? What renders in the template as a result?